test/test.js

```diff
@@ -76,28 +78,56 @@ describe('JUNIT Reporter', () => {
       assert.ok(anotherFile.$.errors === '1');
       assert.ok(anotherFile.testcase.length === parseInt(anotherFile.$.errors, 10));
 
-      assert.ok(testFile.testcase[0].failure[0]._ === 'line 4, col 5, The universal selector shouldn\'t be used. (universalSelector)');
-      assert.ok(testFile.testcase[1].failure[0]._ === 'line 5, col 17, Unit should not be omitted on zero values. (zeroUnit)');
-      assert.ok(anotherFile.testcase[0].failure[0]._ === 'line 6, col 9, Property ordering is not alphabetized (propertyOrdering)');
+      assert.ok(testFile.testcase[0].failure[0]._.trim() === 'line 4, col 5, The universal selector shouldn\'t be used. (universalSelector)');
+      assert.ok(testFile.testcase[1].failure[0]._.trim() === 'line 5, col 17, Unit should not be omitted on zero values. (zeroUnit)');
+      assert.ok(anotherFile.testcase[0].failure[0]._.trim() === 'line 6, col 9, Property ordering is not alphabetized (propertyOrdering)');
 
-      assert.ok(exitCode === 1);
-      exitCode = null;
       done();
     });
   });
 
   it('It should not throw an exit code when no errors', (done) => {
+    sinon.stub(process, 'exit');
     const inspect = stdout.inspect();
 
     junitReporter.report([]);
 
     inspect.restore();
+    // Start by checking the exit code so we can restore process.exit
+    assert.ok(!process.exit.called);
+    process.exit.restore();
 
     const response = inspect.output;
 
     assert.ok(response.length === 0);
 
-    assert.ok(exitCode === null);
+    done();
+  });
+
+  it('It should be able to write to file', (done) => {
+    sinon.stub(process, 'exit');
+    const filename = 'junit.xml';
+    const inspect = stdout.inspect();
+
+    junitReporter.report(exampleResults);
+    const output = junitReporter.out[0];
+
+    const writeOut = junitReporter.writeOut(filename);
+    writeOut();
+
+    inspect.restore();
+    process.exit.restore();
+
+    // Wait 100 ms before checking if file exists, as writeOut is asynchronous
+    setTimeout(() => {
```

Comment:
Shouldn't we use a promise here if asynchronous, instead of a timeout?

Reply:
Of course. I don't know why I didn't do that originally. I have updated the test.